src/fault_injector/fault_types/fault.py

```diff
@@ -3,18 +3,33 @@
 from src.fault_injector.fault_configurations.fault_configuration import (
     FaultConfiguration,
 )
+from src.interlocking_component.route_controller import IInterlockingDisruptor
 from src.logger.logger import Logger
+from src.wrapper.simulation_object_updating_component import (
+    SimulationObjectUpdatingComponent,
+)
+from src.wrapper.simulation_objects import Track, Train
 
 
 class Fault(ABC):
     """An abstract fault for the fault injection"""
 
-    configuration: "FaultConfiguration"
+    configuration: FaultConfiguration
     logger: Logger
+    wrapper: SimulationObjectUpdatingComponent
+    interlocking: IInterlockingDisruptor
 
-    def __init__(self, configuration, logger: Logger):
+    def __init__(
+        self,
+        configuration,
+        logger: Logger,
+        wrapper: SimulationObjectUpdatingComponent,
```

Comment:
I'm unsure if "wrapper" is the best name for this attribute 🙃 What do you think about `simulation_object_updater`?

Reply:
`wrapper` is closer to the component scheme I use for the other components, but you are right, `simulation_object_updater` is closert to the actual classes!